### PR TITLE
feat(ui): Reduce and refresh the `batch_size` growing sync-mode of `invites` in `RoomListService`

### PR DESCRIPTION
**Pull request by Hywan**
This PR can be reviewed commit-by-commit.

The first patch reduces the `batch_size` of `invites` from 100 to 20.

The second does the following.
Inside `RoomListService`, the `State` enum handles the transition from
one state to another. In case of some `State::Error`, the `all_rooms`
sliding sync was refreshed, i.e. its sync-mode was reset to its initial
value.

This patch also refreshes the `invites` sliding sync list! It adds
the `ResetInvitesListGrowingSyncMode` action, and attaches it to the
`refresh_lists` actions.

New constants are added to represent default `batch_size` values for the
growing sync-mode for various sliding sync list. It could be helpful for
further maintenance.

This patch finally adds and updates the tests accordingly.

---

* Fixes https://github.com/vector-im/element-x-ios/issues/1509

### CODE REVIEW

File: crates/matrix-sdk-ui/src/room_list_service/state.rs [hunk 1]
```diff
@@ -177,6 +187,26 @@ impl Action for AddInvitesList {
...
+        sliding_sync
+            .on_list(INVITES_LIST_NAME, |list| {
+                list.set_sync_mode(SlidingSyncMode::new_growing(
+                    INVITES_DEFAULT_GROWING_BATCH_SIZE,
+                ));
+
+                ready(())
+            })
+            .await
+            .ok_or_else(|| Error::UnknownList(ALL_ROOMS_LIST_NAME.to_owned()))?;
```
Comment (bnjbvr) [inline crates/matrix-sdk-ui/src/room_list_service/state.rs:204]: nit: invite list name here

**Review by bnjbvr — APPROVED** [review]
Nice!

**Comment by ara4n** [comment]
@Hywan would it be possible to also defer starting to grow the invite list until after the first batch of visible_rooms has come through? otherwise it will double the amount of time to make visible_rooms visible (as it'll load 40 rooms rather than 20 rooms, for folks with >20 invites).

File: crates/matrix-sdk-ui/src/room_list_service/state.rs (hunk 1 above)
Comment (Hywan) [reply crates/matrix-sdk-ui/src/room_list_service/state.rs]: Good eyes! Thanks.

**Comment by Hywan** [comment]
@ara4n I need to think about this.

1. Element X is using `State::SettingUp` to know when `visible_rooms` and `invites` are available. Introducing a new state means propagating changes up to Element X. Not a big deal, but it has implications, and I need to check with their respective tech leads.
2. When opening the app (first launch) from an invite notification, we would need 3 sync roundtrips (between the client and the server) to get something displayable for the user. I'm not sure that's an improvement.

Honestly, fetching 20 invites shouldn't slow anything down. It's quite a small number of rooms, esp. for invites, where there is no timeline. Let's first merge this PR, and let's measure and see if we can improve things further. Plus, it's also really rare to have more than 20 pending invites ;-). I'm OK to optimise for edge cases but not at the detriment of common cases (cf. point 2). As an alternative, `invites` can have multiple sync-modes, _à la_ `all_rooms`:

* `State::Init`: `invites` is absent (giving the priority to `all_rooms`)
* `State::SettingUp`: `invites` is added in selective sync-mode with a range of `0..=5`, quite a really small range, even `0..=2` would be fine to display the latest invites.
* `State::Running`: `invites` switches to growing sync-mode with a batch size of `20` or something bigger.

Anyway. Let's talk about this, measure, and see how things can be improved, thoughts?

**Comment by ara4n** [comment]
okay, let's measure and see how good/bad it is. it's worth noting that currently it is common for abusive invites to stack up, given there's no way to block invites to a given room (only from a given user).

**Pull request merged**